### ISSUE TITLE
Adding column with type ENUM fails when schema is not main

### ISSUE THREAD

**Issue by Admolly** [duckdb/duckdb#3076]
#### What happens?
The following sequence of commands:
```
CREATE SCHEMA not_main
CREATE TYPE enum_type AS ENUM('a','b','c')
CREATE TABLE not_main.tbl (col1 INT)
ALTER TABLE not_main.tbl ADD COLUMN col2 enum_type
```

produces the following error:
> RuntimeError: Not implemented Error: DataType enum_type not supported yet...a

However, the following does work:
```
CREATE TABLE main.tbl (col1 INT)
ALTER TABLE main.tbl ADD COLUMN col2 enum_type
```
#### To Reproduce
See above.

#### Environment (please complete the following information):
 - OS: Windows 10
 - DuckDB Version: 0.3.2-dev1201
 - DuckDB Client: Python 

#### Before Submitting

- [ ] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [X] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

**Comment by arjenpdevries** [comment]
Shouldn't the type be created in the other schema?

**Comment by Admolly** [comment]
perhaps, but it seems strange to me that a type would be specific to a schema.  What is the proper syntax for specifying which schema a type belongs to?  I don't see anything mentioned in the documenation.

**Comment by arjenpdevries** [comment]
I see! Simply prepending the schema name to the type name raises a parser error. I tried a few variants but I have indeed not managed to use an ENUM type in a different schema than main;
also not by creating the ENUM after setting the schema using SET SCHEMA.

(I could also not find how the ENUM types are represented in the catalog, it seems there is no SQL level catalog info accessible for these and you'd have to use a CPP program to get access to that info - unless I overlooked a table still.)
